tests/observers/k8s_api_test.py

```diff
@@ -3,21 +3,22 @@
 import pytest
 import yaml
 from tests.helpers.assertions import has_datapoint
-from tests.helpers.kubernetes.utils import add_pod_spec_annotations
+from tests.helpers.kubernetes.utils import add_pod_spec_annotations, get_discovery_rule
 from tests.helpers.util import wait_for
 from tests.paths import TEST_SERVICES_DIR
 
 
 @pytest.mark.kubernetes
 def test_k8s_api_observer_basic(k8s_cluster):
     nginx_yaml = TEST_SERVICES_DIR / "nginx/nginx-k8s.yaml"
+    discovery_rule = get_discovery_rule(nginx_yaml, "k8s-api", namespace=k8s_cluster.test_namespace)
```

Comment:
Not really a big fan of the `get_discovery_rule` helper because it obscures what it's actually testing in terms of discovery rules.  Does the helper do something needed for tests to consistently pass?

Reply:
> Not really a big fan of the `get_discovery_rule` helper because it obscures what it's actually testing in terms of discovery rules. Does the helper do something needed for tests to consistently pass?

I added `get_discovery_rule` just to include more endpoint variables and the `Get` and `Contains` functions in the test.

Reply:
It would be better to have tests that explicitly test those things, e.g. `test_pod_labels_in_discovery_rule` or something so that it was explicit and obvious.